### PR TITLE
#167323728 user should be able to receive in app notifications

### PR DESCRIPTION
**Pull request by timi-codes**
#### Description
 - Users can get in-app notification on being followed 
 - An Author can get in-app notification when there is a new comment on their article.
 - All followers of an author get a notification when they publish a new article
#### Type of Change

- [x] New feature (Non-breaking change which adds functionality)

- [x]  Bug fix (Non-breaking change which adds functionality)

#### How Has This Been Tested?
  Integration Testing
#### Pivotal Stories
[#167323728](https://www.pivotaltracker.com/story/show/167323728)

### CODE REVIEW

File: src/helpers/notifications.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    id: followeeId
+  });
+
+  const { id: receiverId, emailNotify } = receiver;
+
+  const message = `<b>${sender.username}</b> just followed you.`;
+  const link = `/profile/${sender.username}`;
+
+  if (emailNotify) {
+    // TODO: send email notification at this point
```
Comment (kechyy) [inline src/helpers/notifications.js]: Remove this emailNotify condition if not needed.

**Review by kechyy — CHANGES_REQUESTED** [review]
Good Job @timi-codes and @codexempire . I think I like the test break down. Was able to understand the implementation from the test. And the test is not dependent on the app. However, I left just a comment you may want to quickly attend to.

````|Well done.

**Review by kechyy — APPROVED** [review]

**Review by tolumide-ng — APPROVED** [review]
This looks awesome 👍 Good job @timi-codes  and @codexempire

**Review by Onyimatics — APPROVED** [review]
Good work @timi-codes and @codexempire.

File: src/test/notification.spec.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+      'Email notifications has been enabled'
+    );
+  });
+  it('Returns 200 if notificationhas been disabled', async () => {
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+    newUser.emailNotify = true;
+    const response = await chai
+      .request(app)
+      .patch(`/api/v1/notifications/opt`)
```
Comment (segunolalive) [inline src/test/notification.spec.js:279]: I think this should be `api/v1/users/:id/notifications`

File: src/test/notification.spec.js [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+  });
+});
+
+describe('GET /api/v1/notifications', () => {
+  beforeEach(async () => {
+    await User.destroy({ truncate: true, cascade: true });
+    await Follower.destroy({ truncate: true, cascade: true });
+    await Notification.destroy({ truncate: true, cascade: true });
+  });
+  it('Returns 404 and error message if no notification was found', async () => {
```
Comment (segunolalive) [inline src/test/notification.spec.js]: `404` means the route/resource was not found.

File: src/test/notification.spec.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  onCommentNotification,
+  onPublishArticleNotification,
+  createInAppNotification
+} = NotificationHelper;
+
+config();
+chai.use(chaiHttp);
+
+const { User, Follower, Notification } = db;
+describe('Test notification helper functions', async () => {
```
Comment (segunolalive) [inline src/test/notification.spec.js]:
```
describe('Notification helper functions', () => {
   describe('CreateInAppNotification', () => {
    it('returns an object of the created notification', () =>{
    . . .
    } ) 
  }
})
```

File: src/controllers/notification.controller.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    try {
+      const notification = await BaseRepository.findOneByField(Notification, {
+        id: notificationId,
+        receiverId
+      });
+
+      if (notification) {
+        const notificationRead = notification.read;
+        if (notificationRead === true) {
+          return responseGenerator.sendError(
```
Comment (segunolalive) [inline src/controllers/notification.controller.js]: `sendSuccess`

**Review by segunolalive — CHANGES_REQUESTED** [review]
Great job 💪 
Fix the errors. Merge @codexempire's corrections

@tolumide-ng review and merge, if satisfactory.

**Pull request merged**